### PR TITLE
Refactor code to follow rust idioms better

### PR DESCRIPTION
**Pull request by dralley**
Fixed some things suggested by ```cargo clippy```

Moved some tests around to follow the standard test organization
https://doc.rust-lang.org/book/ch11-03-test-organization.html

Added a lib.rs file

### CODE REVIEW

**Comment by Richterrettich** [comment]
Hi and thanks for the PR! I'll have a look

File: src/rpm.rs [hunk 1]
```diff
@@ -1721,4 +1721,277 @@ impl RPMBuilder {
...
+mod tests {
+    use super::*;
+    use std::io;
+    use std::io::prelude::*;
+
+    #[test]
+    fn test_header() -> Result<(), Box<dyn std::error::Error>> {
+        let d = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let mut rpm_file_path = d.clone();
+        rpm_file_path.push("tests/assets/389-ds-base-devel-1.3.8.4-15.el7.x86_64.rpm");
```
Comment (Richterrettich) [inline src/rpm.rs]: you are referencing a file that does not exist. You've removed test_assets/389-ds-base-devel-1.3.8.4-15.el7.x86_64.rpm but you did not add it to the tests directory

File: src/rpm.rs [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
-
-use sha1;
-use sha2;
-use sha2::Digest;
 use std::collections::{BTreeMap, BTreeSet};
 use std::convert::TryInto;
 use std::fmt;
 use std::fmt::Display;
 use std::io::{Read, Write};
+use std::str::FromStr;
```
Comment (Richterrettich) [inline src/rpm.rs]: This is a duplicate import

**Comment by Richterrettich** [comment]
I think standardizing the project layout is a good thing and I want to merge your changes as soon as my findings are resolved. I have to coordinate this with #4 though.

**Comment by dralley** [comment]
Hi -- sorry about the missing file, it was done right in a previous commit and I decided to re-do it and must have missed it.

Totally understand w/ the other PR.  Mine is probably easier to rebase and I'm fine with doing so when theirs is finished.

**Comment by Richterrettich** [comment]
So, #4 got merged. I'm open for your PR now.

**Comment by drahnr** [comment]
@dralley I am sorry that a lot of things got shifted a bit, if this becomes too frustrating please ping me, and then let's see how to get this moving :)

**Comment by dralley** [comment]
No problem.  I'm going to reduce the scope of the PR a bit though.  Creating lib.rs and moving the tests around required changing visibility of things and adjusting the API in ways that I don't really  feel comfortable doing as a drive-by contributor.

**Comment by dralley** [comment]
I'm done with this for now FYI.  Feel free to go ahead and merge it as-is.  If I come back to it eventually I'll just make a new PR.

**Comment by Richterrettich** [comment]
This looks good to me.

**Pull request merged**